CRM/Admin/Form/Setting.php

```diff
@@ -101,9 +102,23 @@ public function buildQuickForm() {
       $props = $settingMetaData['values'][$setting];
       if (isset($props['quick_form_type'])) {
         if (isset($props['pseudoconstant'])) {
-          $options = civicrm_api3('Setting', 'getoptions', array(
-            'field' => $setting,
-          ));
+          if (array_key_exists('optionGroupName', $props['pseudoconstant'])) {
```

Comment:
getoptions should do this - & if it doesn't get label then I feel like we should address that (possibly as an option on the call)

Reply:
(although you could probably settle for name for this PR & punt that)